pkg/controllers/dynakube/extension/eec/reconciler.go

```diff
@@ -70,3 +84,105 @@ func (r *reconciler) Reconcile(ctx context.Context) error {
 
 	return r.createOrUpdateStatefulset(ctx)
 }
+
+func (r *reconciler) reconcileTLSSecret(ctx context.Context) error {
+	query := k8ssecret.Query(r.client, r.client, log)
+
+	secret, err := query.Get(ctx, types.NamespacedName{
+		Name:      getSelfSignedTLSSecretName(r.dk.Name),
+		Namespace: r.dk.Namespace,
+	})
+
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return err
+	}
+
+	if r.dk.Spec.Templates.ExtensionExecutionController.TlsRefName != "" {
+		if secret != nil {
```

Comment:
The secret is also not nil if err == NotFound. In that case the delete fails, because the resource name is empty.
```suggestion
		if secret != nil && !k8serrors.IsNotFound(err) {
```

Reply:
Refactored to this (we only get no error when the secret exists)
```
if err == nil
```